ipatests/test_integration/test_installation.py

```diff
@@ -703,6 +703,29 @@ def install(cls, mh):
     def test_install_master(self):
         tasks.install_master(self.master, setup_dns=False)
 
+    def test_smoke_test_for_debug_mode(self):
+        """Test if an IPA server works in debug mode.
+        Related: https://pagure.io/freeipa/issue/8891
+        """
+
+        target_fname = paths.IPA_SERVER_CONF
+        assert not self.master.transport.file_exists(target_fname)
+
+        # set the IPA server in debug mode
+        server_conf = "[global]\ndebug=True"
+        self.master.put_file_contents(target_fname, server_conf)
+        self.master.run_command([paths.IPACTL, "restart"])
```

Comment:
The full restart of `ipa` takes ~1-2min, while the restart of `httpd` takes several seconds. We waste here ~2-4min.

Reply:
Less than that if I can trust the [log](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/8848656a-d58d-11eb-99d1-fa163ed6a638/runner.log.gz) : 2x50s, so 1m40. It also makes sure the whole of ipa comes up in debug mode (bonus!). I find this much better than having to detect the running platform's httpd service (or making sure it uses systemd...).

Reply:
> Less than that if I can trust the [log](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/8848656a-d58d-11eb-99d1-fa163ed6a638/runner.log.gz) : 2x50s, so 1m40. 

Azure log:
https://dev.azure.com/freeipa/freeipa/_build/results?buildId=9187&view=logs&j=fd5ef46c-a89d-5fee-07c3-d9f8082531c5&t=e993e2cc-4ff6-5b17-419c-da6d7496f246&l=844

> It also makes sure the whole of ipa comes up in debug mode (bonus!). I find this much better than having to detect the running platform's httpd service (or making sure it uses systemd...).

IPA doesn't support non-systemd platforms...

> Plus, ipactl is the abstraction for the platform, so it works everytime.

Reply:
> IPA doesn't support non-systemd platforms...

For now? FreeIPA went from non-systemd to systemd, who knows what the future holds. 
Anyway, I'd rather restart than having to detect how the httpd service is named on the targed host.

Reply:
May be https://pagure.io/freeipa/issue/8620 more clearly explains what I'm trying to achieve.
If that PR will be merged then every service name will be resolved on a remote ipaplatform.
I don't see any problem to do it manually in this PR.


> Anyway, I'd rather restart than having to detect how the httpd service is named on the targed host.

Why?

Reply:
I also think that a full restart is overkill and this should revert to systemctl restart httpd.

While I agree in principal that service handling should be abstracted to not be system-dependent I don't think that it should block this individual change, particularly since it is a regression with older python releases This will just represent one more place to fix on that abstraction is in place.

Reply:
> May be https://pagure.io/freeipa/issue/8620 more clearly explains what I'm trying to achieve.
> If that PR will be merged then every service name will be resolved on a remote ipaplatform.
> I don't see any problem to do it manually in this PR.

I appreciate what you are trying to achieve, but either it is done globally or not. 

> > Anyway, I'd rather restart than having to detect how the httpd service is named on the targed host.
> 
> Why?

Because "ipactl restart" works on all supported platforms.

Reply:
> I also think that a full restart is overkill and this should revert to systemctl restart httpd.
> 
> While I agree in principal that service handling should be abstracted to not be system-dependent I don't think that it should block this individual change, particularly since it is a regression with older python releases This will just represent one more place to fix on that abstraction is in place.

@rcritten @stanislavlevin please let me know what you would like best: "ipactl restart", which is platform-independent and a complete smoke test (e.g. can we start and restart ipa in debug mode) but (I agree) slow, or "systemctl restart httpd" which is much faster but platform-dependent.

Reply:
Neither of these. I proposed and propose resolve service name + systemctl restart resolved_name
The example of code for resolving _remote_ name is above (btw, copy-pasted from current IPA code).

Reply:
In current IPA code there are many places with the same code snippet for resolving remote service name.

Reply:
Flagging postponed.